### PR TITLE
Refactor BS to merge various threads into one.

### PR DESCRIPTION
**Pull request by kormat**
BeaconServer.worker() thread covers:
- Reading from shared cache on ZK
- Getting ZK lock
- Propagating PCBs
- Registering paths  

(Depends on #339)
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/342%23discussion_r37792459%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/342%23issuecomment-134346794%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/342%23discussion_r37843107%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/342%23discussion_r37844874%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/342%23discussion_r37845224%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/342%23issuecomment-134346794%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22lgtm%22%2C%20%22created_at%22%3A%20%222015-08-24T19%3A23%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20d6e4a7975fd823f6d8e4591c8410b457b9974aa2%20lib/zookeeper.py%2023%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/342%23discussion_r37792459%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20not%20%60_conn_epoch%60%20%3F%22%2C%20%22created_at%22%3A%20%222015-08-24T19%3A20%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Because%20other%20objects%20may%20need%20to%20reference%20it%2C%20to%20see%20if%20their%20own%20state%20is%20stale.%22%2C%20%22created_at%22%3A%20%222015-08-25T08%3A24%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22For%20example%2C%20%60ZkSharedCache.handle_shared_entries%60%20uses%20it%22%2C%20%22created_at%22%3A%20%222015-08-25T08%3A47%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-08-25T08%3A52%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/zookeeper.py%3AL128-140%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/342#issuecomment-134346794'>General Comment</a></b>
- <a href='https://github.com/pszalach'></a> lgtm
- [x] <a href='#crh-comment-Pull d6e4a7975fd823f6d8e4591c8410b457b9974aa2 lib/zookeeper.py 23'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/342#discussion_r37792459'>File: lib/zookeeper.py:L128-140</a></b>
- <a href='https://github.com/pszalach'></a> why not `_conn_epoch` ?
- <a href='https://github.com/kormat'></a> Because other objects may need to reference it, to see if their own state is stale.
- <a href='https://github.com/kormat'></a> For example, `ZkSharedCache.handle_shared_entries` uses it

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/342?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/342?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/342'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: lib/zookeeper.py [hunk 1]
```diff
@@ -129,10 +128,12 @@ def __init__(self, isd_id, ad_id, srv_type, srv_id,
         self._lock = threading.Event()
         # Used to signal connection state changes
         self._state_events = queue.Queue()
+        self.conn_epoch = 0
```
Comment (pszal) [inline lib/zookeeper.py]: why not `_conn_epoch` ?

Comment (kormat) [inline lib/zookeeper.py]: Because other objects may need to reference it, to see if their own state is stale.

Comment (kormat) [inline lib/zookeeper.py]: For example, `ZkSharedCache.handle_shared_entries` uses it

Comment (pszal) [inline lib/zookeeper.py]: :+1:

**Pull request merged**